### PR TITLE
add types for clevertap-web-sdk

### PR DESCRIPTION
**Pull request by droyson**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/clevertap-web-sdk/test/clevertap-global-tests.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/// <reference types="../index" />
```
Comment (peterblazejewicz) [inline types/clevertap-web-sdk/test/clevertap-global-tests.ts:1]:
```suggestion
```
either use proper import (`import clevertap = require('.....');`, or remove reference in tests. If global script version used, it does not require explicit reference in tests

File: types/clevertap-web-sdk/clevertap.d.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    getTotalVisits(): number | undefined;
+    getLastVisit(): Date | undefined;
+}
+
+interface Session {
+    getTimeElapsed(): number | undefined;
+    getPageCount(): number | undefined;
+}
+
+declare const clevertap: {
```
Comment (peterblazejewicz) [inline types/clevertap-web-sdk/clevertap.d.ts:94]: this looks wrong, this is a class symbol:
https://github.com/CleverTap/clevertap-web-sdk/blob/master/src/clevertap.js#L29
it should support `instanceof` narrowing and other language features

**Review by Shahaed — APPROVED** [review]

File: types/clevertap-web-sdk/test/clevertap-global-tests.ts (hunk 1 above)
Comment (droyson) [reply types/clevertap-web-sdk/test/clevertap-global-tests.ts]: Wasn't aware of this. Thanks 👍

File: types/clevertap-web-sdk/clevertap.d.ts (hunk 2 above)
Comment (droyson) [reply types/clevertap-web-sdk/clevertap.d.ts]: Created an interface of `CleverTap` and used it as the type of `clevertap`. The `CleverTap` class isn't exported from main.js, only an instance is exported. https://github.com/CleverTap/clevertap-web-sdk/blob/master/src/main.js

Comment (peterblazejewicz) [reply types/clevertap-web-sdk/clevertap.d.ts]: Im only pointing from outside onlooker, the code here should look like:
```ts
export default class CleverTap {
```
class could have no constructor defined, as it's not exported later
so the `index.d.ts` (being UMD), could  look like real code in JS module:
```ts
import CleverType from './clevertap';

declare const clevertap: CleverType;
export = clevertap;

export as namespace clevertap;
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@droyson thx!

**Comment by droyson** [comment]
Ready to merge

**Pull request merged**